### ISSUE TITLE
[Question][Self-managed]: Unable to establish Logstash connection with elasticsearch for standalone agent installation.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#576]
**Description:**
We are unable to establish Logstash connection with elasticsearch for standalone agent installation on a self-managed environment.
- We are attempting to install standalone agent and get data through logstash on self-managed 8.3 BC4 environment.

**Kibana Version:**
8.3 BC-4 Self-managed environment.
Platform: Windows 10 x64

**elastic-agent.yml:**
[elastic-agent.zip](https://github.com/elastic/kibana/files/8926442/elastic-agent.zip)

**logstash.conf:**
[logstash.zip](https://github.com/elastic/kibana/files/8926440/logstash.zip)
(Updated as per scenario 1 and 2 shared below)

On running `logstash -f C:\elk\logstash\config\logstash.conf` we are getting below errors:

**Scenario 1:**
Attempted with: `https://localhost:9200`
![2](https://user-images.githubusercontent.com/77374876/174271889-d8128763-b1b8-4fea-a66e-3f922f6c0f48.JPG)

**Scenario 2:**
Attempted with: `https://machine-ip:9200`
![1](https://user-images.githubusercontent.com/77374876/174271914-e3ff7e8d-73c1-4f2e-a29a-ab217cb4d199.JPG)

**Elasticsearch:**
![3](https://user-images.githubusercontent.com/77374876/174271965-221c8994-8322-44b0-af75-a4cdb85639e2.JPG)

**Impacted Testcase:**
[T1192851](https://elastic.testrail.io/index.php?/tests/view/1192851)

**Comment by amolnater-qasource** [comment]
Hi Team
We have revalidated running standalone agent through logstash on 8.4 BC2 self-managed environment.
- We are still getting same errors on running logstash.

**Screenshots:**
![6](https://user-images.githubusercontent.com/77374876/182818150-14d30004-671c-4dbe-bb1a-e5ebba830fa6.JPG)

Currently this is impacting our execution of testcase at [T1541622](https://elastic.testrail.io/index.php?/tests/view/1541622&group_by=cases:section_id&group_order=asc&group_id=30365)

Could anyone please confirm if there is any update in the steps to be followed for running standalone agent through logstash.

Please let us know if we are missing anything here.
Thank you

**Comment by joshdover** [comment]
@jsvd do you have any advice here on what could be going wrong? I don't think there's a specific issue with Agent's connection to Logstash but rather some adjustments that need to be made to allow Logstash to trust ES's cert, probably with the [`cacert` or `ca_trusted_fingerprint` options](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-cacert).

However, these errors around a valid cert path are a bit confusing

**Comment by jsvd** [comment]
The error message "unable to find valid certification path to requested target" means that Logstash tried to validate a certificate chain by taking the certificate given by Agent + whatever certificates can be found in the files passed to in [ssl_certificate_authorities](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-elastic_agent.html#plugins-inputs-elastic_agent-ssl_certificate_authorities), but failed to construct a valid chain. We can grab more debugging by passing a jvm flag `LS_JAVA_OPTS="-Djavax.net.debug=all"`.

The log entries also mention something along the lines of "invalid protocol", which typically means that whatever is connecting to the tcp port isn't speaking the lumberjack protocol.
It'd be nice to get full logging files to help see the sequence of error messages, and even better if you can produce the issue with a set of throwaway certificates that we can recreate on our side it'd help a lot.

**Comment by amolnater-qasource** [comment]
Hi @joshdover 
Thank you for looking into this and sharing the feedback.

We will be revalidating this on latest self-managed 8.5 SNAPSHOT with cacerts.

Thanks!

**Comment by amolnater-qasource** [comment]
Hi Team
We have revalidated this issue on latest 8.5 SNAPSHOT self-managed environment using cacerts and now we are successfully able to establish Logstash connection with elasticsearch for standalone agent installation. 
- Standalone agent data is available under Data Streams tab when Logstash output is selected.

**Build details:**
VERSION: 8.5.0
BUILD: 55953
COMMIT: 0f09b0eea3201a30896c2a1b22266028351ce512

**Screenshot:**
![9](https://user-images.githubusercontent.com/77374876/188381082-2f485b17-7d98-473d-bda2-92b344cdc852.JPG)

Further we have updated **01** testcase under Fleet test suite at link: 
- [[Self-managed]: Validate user is able to see data for standalone agent running through logstash.](https://elastic.testrail.io/index.php?/cases/view/167099)

Hence we are closing this and marking as **QA:Validated**.

Thanks!
